### DISCUSSION TITLE
How to enable vue devtools

### DISCUSSION THREAD

**Discussion by ivramartono** [nuxt/nuxt#16237] [Questions]
I am hosting my nuxt app v3.0.0-rc.9 on an nginx server statically using nuxt generate. I am trying to enable Vue devtools plugin on a testing/production environment by changing the vue config in nuxt.config.ts: 
```
dev: true,
vue: {
   config: {
      productionTip: false,
      devTools: true,
   },
},
```
As well as changing my .env variables to 
```
NODE_ENV=development
__VUE_PROD_DEVTOOLS__=true

```
However I am still getting `Nuxt.js + Vue.js is detected on this page. Devtools inspection is not available because it's in production mode or explicitly disabled by the author.`

**Comment by hoanghadu** [comment]
I get the same issue. Have you found any solution?

**Comment by Baroshem** [comment]
I would not recommend enabling dev tools for testing / production environment. Dev Tools give access to the state of the components and pages that can leak some data. Could you expand your use case? 

Maybe there is another way to solve it without exposing dev tools on prod env?

**Reply by TheAlexLichter** [reply]
I don't share the concerns there - You can enable devtools easily on client-side for every production page and all the JS is human-readable anyway 🙈

**Reply by Baroshem** [reply]
Sure but you dont have access to a reactive state that easily. Meaning that making changes to how your app is behaving is more difficult in the production code that in the reactive vue tools. Imaging having a global reactive value `isAdmin`. Changing this in vue dev tools is really simple to see the new layout for admin. You would just search for the page and toggle the value. While in the production code you would have to scan a lot of regular JS to see it.

**Reply by cas-technology** [reply]
The usecase is definitely valid for testing environments, where the devtools could be used to see or modify state. 

The argument that it prevents reactive values being modified is moot. All frontend code is visible and open for modification so having the _ability_ for devtools to be _enabled_ on perhaps unintended environments (such as testing or staging) via the _intended_ way as was previously described in the docs for Nuxt 2 is in my opinion very much welcome.
